### PR TITLE
BV: Added support for bit-vectors in the yices wrapper

### PR DESCRIPTION
**Pull request by mikand**

### CODE REVIEW

File: pysmt/solvers/yices.py [hunk 1]
```diff
@@ -243,20 +247,32 @@ def _to_term_array(self, arr):
         #pylint: disable=star-args
         return (libyices.term_t * len(arr))(*arr)
 
+    def _check_term_result(self, res):
+        if res == -1:
+            err = str(libyices.yices_error_string())
+            assert False, "Yices returned an error: " + err
```
Comment (marcogario) [inline pysmt/solvers/yices.py]: raise InternalSolverError(...)

File: pysmt/solvers/yices.py [hunk 2]
```diff
@@ -243,20 +247,32 @@ def _to_term_array(self, arr):
...
+        res = libyices.yices_not(args[0])
+        self._check_term_result(res)
+        return res
 
     def walk_symbol(self, formula, **kwargs):
         symbol_type = formula.symbol_type()
         var_type = self._type_to_yices(symbol_type)
         term = libyices.yices_new_uninterpreted_term(var_type)
         libyices.yices_set_term_name(term, String(formula.symbol_name()))
+        assert term != -1
```
Comment (marcogario) [inline pysmt/solvers/yices.py]: ```self._check_term_result(term)```

File: pysmt/solvers/yices.py [hunk 3]
```diff
@@ -348,8 +396,125 @@ def walk_function(self, formula, args, **kwargs):
...
+        if width <= 64:
+            # we can use the numberical representation
+            value = formula.constant_value()
+            res = libyices.yices_bvconst_uint64(width, value)
+        else:
+            # we must resort to strings to communicate the result to yices
+            short_rep = str(bin(formula.constant_value()))[2:]
+            if formula.constant_value() < 0:
+                raise NotImplementedError
+            bin_str = short_rep.rjust(width, "0")
```
Comment (marcogario) [inline pysmt/solvers/yices.py]: FNode.bv_bin_str ?

File: pysmt/solvers/yices.py [hunk 4]
```diff
@@ -348,8 +396,125 @@ def walk_function(self, formula, args, **kwargs):
...
+        self._check_term_result(res)
+        return res
+
+    def walk_bv_mul(self, formula, args, **kwargs):
+        res = libyices.yices_bvmul(args[0], args[1])
+        self._check_term_result(res)
+        return res
+
+    def walk_bv_udiv(self, formula, args, **kwargs):
+        res = libyices.yices_bvdiv(args[0], args[1])
```
Comment (marcogario) [inline pysmt/solvers/yices.py]: Is this the unsigned one?

Comment (mikand) [inline pysmt/solvers/yices.py]: Yes, the signed one is ```libyices.yices_bvsdiv``` (similarly for rem)

**Pull request merged**